### ISSUE TITLE
`--number-sections` option causes "unlisted" sections to appear in TOC

### ISSUE THREAD

**Issue by fbristow** [jgm/pandoc#6339]
Summary
========

I want to use the `.unlisted` class on some of my headings so that they don't appear in the TOC, however when I use the `--number-sections` command-line option, that class is not respected.

pandoc version: 2.9.2.1

Sample input
===========

```markdown
First {.unnumbered .unlisted}
=====

Second {.unnumbered}
======

Third
=====
```

Command used
============

`pandoc input.md --toc --standalone --number-sections -t html`

Actual output
==========

**Note**: Header is omitted for brevity.

```html
<body>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#first">First</a></li>
<li><a href="#second">Second</a></li>
<li><a href="#third"><span class="toc-section-number">1</span> Third</a></li>
</ul>
</nav>
<h1 class="unnumbered unlisted" data-number="" id="first">First</h1>
<h1 class="unnumbered" data-number="" id="second">Second</h1>
<h1 data-number="1" id="third"><span class="header-section-number">1</span> Third</h1>
</body>
```

Expected output
============

**Note**: Header is omitted for brevity.

```html
<body>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#second">Second</a></li>
<li><a href="#third"><span class="toc-section-number">1</span> Third</a></li>
</ul>
</nav>
<h1 class="unnumbered unlisted" data-number="" id="first">First</h1>
<h1 class="unnumbered" data-number="" id="second">Second</h1>
<h1 data-number="1" id="third"><span class="header-section-number">1</span> Third</h1>
</body>
```

**Comment by jgm** [comment]
Looking  the discussion at #1762 helps.  It turns out that it's problematic to combine numbering with `unlisted` in certain formats, so we only make `unlisted` have any effect when `unnumbered` is also used -- and this is documented.  So, not a bug.

**Comment by jgm** [comment]
Solution for you is to use `{.unlisted .unnumbered}`.

**Comment by fbristow** [comment]
Thanks for your quick response, @jgm.

Using the other order (`{.unlisted .unnumbered}`) produces the same result (header appear in TOC when `--number-sections` is enabled).

Here's my story: I started writing a document with `--number-sections` and with some headers having `.unnumbered`. Using `.unnumbered` on a section header, to me, implies that the document *will* be numbered.

I discovered `.unlisted` *via* #1762, and after reading it (admittedly not very carefully), tried to use it knowing that it would only work with `.unnumbered`, but I couldn't see any effect in my document. 

On minimizing my sample to submit the bug, I figured out that `.unlisted` is negated by `--number-sections`, which I found pretty confusing.

Long story short, even if it's not a bug, it feels like a papercut.

FWIW: After *re*reading #1762 and thinking a bit more, what I'm really looking for is an unlisted appendix in HTML...

**Comment by jgm** [comment]
OK, if that's how it behaves then it is a bug indeed.

**Comment by jgm** [comment]
Now properly fixed.
